src/prop-types/tools/areComponentsEqual.js

```diff
@@ -0,0 +1,12 @@
+const RHL_PREFIX = '__reactstandin__getCurrent';
+
+const getProxyOrType = c => {
+  const isProxy = c && typeof c !== 'string' && Reflect.has(c, RHL_PREFIX);
+  return isProxy ? c[RHL_PREFIX]() : c;
+};
+
+const areComponentsEqual = (a, b) => {
```

Comment:
Probably we'd take an array for `b`, or make this function varargs?

Reply:
Yep, we can, but I wouldn't do that in the helper method though. I still prefer this https://github.com/IBM/carbon-components-react/blob/47b7447f18792f8d1655863f75dcf749cec19ce4/src/prop-types/childrenOf.js#L20-L33